consensus/src/block_storage/sync_manager.rs

```diff
@@ -231,17 +232,32 @@ impl BlockStore {
             + 1;
 
         let blocks = retriever
-            .retrieve_block_for_qc(highest_ordered_cert, num_blocks)
+            .retrieve_block_for_qc(
+                highest_ordered_cert,
+                num_blocks,
+                highest_ledger_info.commit_info().id(),
+            )
             .await?;
 
         assert_eq!(
-            blocks.first().expect("should have at least 3-chain").id(),
+            blocks.first().expect("blocks are empty").id(),
             highest_ordered_cert.certified_block().id(),
+            "Expecting in the retrieval response, first block should be {}, but got {}",
+            highest_ordered_cert.certified_block().id(),
+            blocks.first().expect("blocks are empty").id(),
+        );
+
+        assert_eq!(
```

Comment:
why do we need to assert round specifically?

Reply:
good point, previously I thought malicious user could send a sync info with inconsistent round/block_id, but given that it's a qc, the information should be consistent.